Cython/Compiler/ExprNodes.py

```diff
@@ -7649,7 +7779,10 @@ def generate_parallel_assignment_code(self, rhs, code):
                           or not rhs.type.is_builtin_type)
         long_enough_for_a_loop = len(self.unpacked_items) > 3
 
-        if special_unpack:
+        if rhs.type.is_fastcall_tuple:
+            self.generate_fastcall_tuple_parallel_unpacking_code(
```

Comment:
Good point. This looks like another case where it's probably better to fall back to coercion rather than reimplement everything since it doesn't really help the core "pass *args efficiently to another function" benefit.